### PR TITLE
hurl: init at 1.2.0

### PR DESCRIPTION
**Pull request by eonpatapon**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/hurl/default.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+, openssl
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "hurl";
+  version = "1.2.0";
+
+  src = fetchFromGitHub {
+    owner = "Orange-OpenSource";
+    repo = "hurl";
```
Comment (fabaff) [inline pkgs/tools/networking/hurl/default.nix:15]: Opinionated charge

```suggestion
    repo = pname;
```

**Comment by fabaff** [comment]
Related to #81418

File: pkgs/tools/networking/hurl/default.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    rev = version;
+    sha256 = "0hbyqj794pvvfrg6jgz63mih73bnmnvgmwbv705c2238w7wsgk9w";
+  };
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
```
[the remark this answers is not in the document]
Comment (eonpatapon) [reply pkgs/tools/networking/hurl/default.nix:21]: @fabaff not sure to get what you say. It's first time I'm packaging a rust application.

Looking at `buildRustPackage` it requires either cargoSha256, cargoHash, cargoVendorDir, or cargoLock, and `cargoLock` uses a local file not some that can be found in the source as I can tell

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 131103` at 46779356 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/90573ffbe0a55d5f57756bbe57b5c880">hurl</a></li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

File: pkgs/tools/networking/hurl/default.nix (hunk 2 above)
[the remark this answers is not in the document]
Comment (fabaff) [reply pkgs/tools/networking/hurl/default.nix:21]: Upstream added the `Cargo.lock` file after the 1.2.0 was tagged and released. If a later commit is use then it's parts of the source.  

```nix
rustPlatform.buildRustPackage rec {
  pname = "hurl";
  version = "unstable-2021-07-25";

  src = fetchFromGitHub {
    owner = "Orange-OpenSource";
    repo = "hurl";
    rev = "a6fd63419170935db25fb384bf980f663184ef3d";
    sha256 = "xxx";
  };
```

[the remark this answers is not in the document]
Comment (eonpatapon) [reply pkgs/tools/networking/hurl/default.nix:21]: Actually the `Cargo.lock` is included in the 1.2.0 source tree. So it is preferable to just specify the cargoSha256 ?

File: pkgs/tools/networking/hurl/default.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  nativeBuildInputs = [
+    pkg-config
+  ];
+
+  buildInputs = [
+    libxml2
+    openssl
+  ];
+
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/hurl/default.nix:29]: Please comment why tests are disabled.

Comment (eonpatapon) [reply pkgs/tools/networking/hurl/default.nix:30]: Done

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 131103` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>hurl</li>
  </ul>
</details>

**Review by fabaff — APPROVED** [review]
Looks good to me.

**Pull request merged**